azkaban-common/src/main/java/azkaban/project/FlowLoaderFactory.java

```diff
@@ -41,21 +46,42 @@ public FlowLoaderFactory(final Props props) {
   }
 
   /**
-   * Creates flow loader based on manifest file inside project directory.
+   * Creates flow loader based on project YAML file inside project directory.
    *
    * @param projectDir the project directory
    * @return the flow loader
    */
   public FlowLoader createFlowLoader(final File projectDir) throws ProjectManagerException {
-    // Todo jamiesjc: need to check if manifest file exists in project directory,
-    // and create FlowLoader based on different flow versions specified in the manifest file.
-    final String flowVersion = null;
-    if (flowVersion == null) {
-      return new DirectoryFlowLoader(this.props);
-    } else if (flowVersion.equals(Constants.AZKABAN_FLOW_VERSION_2_0)) {
-      return new DirectoryYamlFlowLoader(this.props);
+
+    final File[] projectFileList = projectDir.listFiles(new SuffixFilter(Constants
+        .PROJECT_FILE_SUFFIX));
+    if (projectFileList != null && projectFileList.length > 0) {
```

Comment:
based on https://docs.oracle.com/javase/7/docs/api/java/io/File.html#listFiles(java.io.FileFilter)
returning null means the input dir is not a dir or IO error happens, in this case should we still fall back to DirectoryFlowLoader ?

Reply:
If listFiles returns null, we should throw Exception. I'll change it, thanks!